spark/src/main/scala/geotrellis/spark/TileLayerMetadata.scala

```diff
@@ -157,20 +157,23 @@ object TileLayerMetadata {
     K: (? => TilerKeyMethods[K, K2]) ,
     V <: CellGrid,
     K2: SpatialComponent: Boundable
-  ](rdd: RDD[(K, V)], crs: CRS, scheme: LayoutScheme): (Int, TileLayerMetadata[K2]) = {
+  ](rdd: RDD[(K, V)], crs: CRS, scheme: LayoutScheme, maxZoom: Int = 0): (Int, TileLayerMetadata[K2]) = {
     val (extent, cellType, cellSize, bounds) = collectMetadata(rdd)
-    val LayoutLevel(zoom, layout) = scheme.levelFor(extent, cellSize)
+    val LayoutLevel(zoom, layout) = (if (maxZoom > 0) scheme.asInstanceOf[ZoomedLayoutScheme].levelForZoom(maxZoom)
+      else scheme.levelFor(extent, cellSize))
     val kb = bounds.setSpatialBounds(KeyBounds(layout.mapTransform(extent)))
     (zoom, TileLayerMetadata(cellType, layout, extent, crs, kb))
   }
 
+
   def fromRdd[
     K: GetComponent[?, ProjectedExtent]: (? => TilerKeyMethods[K, K2]),
     V <: CellGrid,
     K2: SpatialComponent: Boundable
-  ](rdd: RDD[(K, V)], scheme: LayoutScheme): (Int, TileLayerMetadata[K2]) = {
+  ](rdd: RDD[(K, V)], scheme: LayoutScheme, maxZoom: Int = 0): (Int, TileLayerMetadata[K2]) = {
     val (extent, cellType, cellSize, bounds, crs) = collectMetadataWithCRS(rdd)
-    val LayoutLevel(zoom, layout) = scheme.levelFor(extent, cellSize)
+    val LayoutLevel(zoom, layout) = (if (maxZoom > 0) scheme.asInstanceOf[ZoomedLayoutScheme].levelForZoom(maxZoom)
```

Comment:
Probably should use an `Option[Int]` type for the maxZoom, so if someone calls the argument with a maxZoom of 0, we can deal with that error differently than if the argument was not there at all.

We wouldn't want to put this `asInstanceOf` in here (`asInstanceOf` is considered Evil in Scala and we should only use it as a last resort).

The maxZoom parameter shouldn't be used with anything but a `tms` layout, which could be detected in the `EtlConfig` code and an error would be issued at that point.

Here we could avoid the `asInstanceOf` by matching

  ```scala
val LayoutLevel(zoom, layout) = {
  maxZoom match {
    case Some(zoom) =>
      scheme match {
         case zoomedLayoutScheme: ZoomedLayoutScheme =>
            zoomedLayoutScheme.levelFor(extent, cellSize)
         case _ => // throw a descriptive error
      }
    case None =>
      scheme.levelFor(extent, cellSize)
  }
}
```